app/controllers/Emergency.scala

```diff
@@ -31,22 +42,98 @@ object Emergency extends Controller with PanDomainAuthActions {
+          unauthorised("Only Guardian email addresses with two-factor auth are supported.", reissueTopic)
         }
       }
       catch {
         case e: CookieSignatureInvalidException =>
-          unauthorised("Invalid existing session, could not log you in.")
+          unauthorised("Invalid existing session, could not log you in.", reissueTopic)
         case e: CookieParseException =>
-          unauthorised("Could not refresh existing session due to a corrupted cookie.")
+          unauthorised("Could not refresh existing session due to a corrupted cookie.", reissueTopic)
       }
     }).getOrElse {
-      unauthorised("No existing login session found, unable to log you in.")
+      unauthorised("No existing login session found, unable to log you in.", reissueTopic)
+    }
+  }
+
+  def requestCookieLink = EmergencySwitchIsOnAction { req =>
+    Ok(views.html.emergency.newCookieIssued())
+  }
+
+  def sendCookieLink = EmergencySwitchIsOnAction { req =>
+
+    val tokenIssuedAt = DateTime.now().getMillis
+    val emailAddress= req.body.asFormUrlEncoded.get("email")(0)
+
+    if (emailAddress.matches("^[a-z]+\\.[a-z]+@(guardian.co.uk|theguardian.com)$")) {
+
+      val token = Random.alphanumeric.take(20).mkString
+
+      val cookieIssue = NewCookieIssue(token, emailAddress,
+        tokenIssuedAt, false)
+
+      try {
+        val userOpt = Scanamo.put[NewCookieIssue](AWS.dynamoDbClient)(loginConfig.tokensTableName)(cookieIssue)
+        val ses = new SES(loginConfig.sesClient, loginConfig)
+        ses.sendCookieEmail(token, emailAddress)
+
+        Ok(views.html.emergency.emailSent())
+      }
+      catch {
+        case e: Throwable => InternalServerError(e.toString())
+      }
     }
+    else {
+      BadRequest("Only guardian email addresses are supported")
+    }
+  }
+
+  def issueNewCookie(userToken: String) = EmergencySwitchIsOnAction { req =>
+
+    val newCookieTopic = "A new cookie has not been created"
+    val issueNewCookieTopic = "New cookie has not been created"
+    val tenMinutesInMilliSeconds = 600000
+
+    val tableName = loginConfig.tokensTableName
+    val tokenOpt: Option[Xor[DynamoReadError, NewCookieIssue]] = Scanamo.get[NewCookieIssue](AWS.dynamoDbClient)(tableName)('id -> s"$userToken")
+    tokenOpt.map {
+      case Left(error) => {
+        Logger.warn(s"Error when reading entry with $userToken from dynamo. A new cookie will not be issued")
+        unauthorised("Checking your access token failed. You will not be issued with a new ", issueNewCookieTopic)
+      }
+      case Right(tokenEntry: NewCookieIssue) => {
+        if (!tokenEntry.used) {
+          val tokenAgeInMilliseconds = DateTime.now().getMillis - tokenEntry.requested
+          if (tokenAgeInMilliseconds > tenMinutesInMilliSeconds) {
+            Unauthorized(views.html.emergency.newCookieFailure("Your link has expired. Could not create a new cookie"))
```

Comment:
Possibly just logging `token [id] is expired` would be useful.  That way if someone queries mine didn't work it be easy to spot why without going into the Dynamo table. I don't think we need to log the username though.